Dockerfile

```diff
@@ -1,12 +1,14 @@
-FROM ubuntu:14.04
+FROM ubuntu:16.04
+MAINTAINER Aleksandar Todorović <aleksandar@r3bl.me>
 
 # Install dependencies
 RUN apt-get update -y && \
-    apt-get install -y openssl python-imaging python-jinja2 python-lxml libxml2-dev libxslt1-dev python-pgpdump spambayes
+    apt-get install -y tor python openssl python-imaging git python-jinja2 python-lxml libxml2-dev libxslt1-dev python-pgpdump spambayes python-pip && \
+    apt-get upgrade -y
 
 # Add code
-WORKDIR /Mailpile
-ADD . /Mailpile
+RUN git clone --recursive https://github.com/mailpile/Mailpile/ Mailpile
```

Comment:
I agree with @aviau . Since we must enforce coerence between Dockerfile and git repository, it's preferable to get all stuff at time. Or, if your really want to "git clone", use a tag or hash please.